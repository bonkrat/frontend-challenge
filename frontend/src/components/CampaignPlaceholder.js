import React from "react";
import { classnames } from "tailwindcss-classnames";

const root = classnames("flex", "flex-col");
const row = classnames("flex", "h-64", "flex-grow", "px-2", "mb-2");
const image = classnames("md:w-1/2", "bg-gray-100", "py-56");
const description = classnames("bg-gray-100", "mx-6", "md:w-1/2", "py-56");
export const campaignHeader = classnames(
  "flex",
  "flex-col",
  "bg-purple-100",
  "w-full",
  "p-16",
  "flex",
  "justify-center",
  "items-center"
);

/**
 * Placeholder for the campaign view while it fetches data.
 */
const CampaignListPlaceholder = () => (
  <div className={root}>
    <div className={campaignHeader}></div>

    <div className={row}>
      <div className={image}></div>
      <div className={description}></div>
    </div>
  </div>
);

export default CampaignListPlaceholder;
