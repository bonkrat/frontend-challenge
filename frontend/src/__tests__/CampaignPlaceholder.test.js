import React from "react";
import { shallow } from "enzyme";
import CampaignPlaceholder from "../components/CampaignPlaceholder";
import toJson from "enzyme-to-json";

it("renders correctly", () => {
  expect(toJson(shallow(<CampaignPlaceholder />))).toMatchSnapshot();
});
