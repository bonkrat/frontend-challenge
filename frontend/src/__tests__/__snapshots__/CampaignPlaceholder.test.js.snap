// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col"
>
  <div
    className="flex flex-col bg-purple-100 w-full p-16 flex justify-center items-center"
  />
  <div
    className="flex h-64 flex-grow px-2 mb-2"
  >
    <div
      className="md:w-1/2 bg-gray-100 py-56"
    />
    <div
      className="bg-gray-100 mx-6 md:w-1/2 py-56"
    />
  </div>
</div>
`;
